include/directx/d3dx12.h

```diff
@@ -4037,6 +4038,277 @@ class CD3DX12_NODE_MASK_SUBOBJECT
+        INITIALIZE_MEMBER_DATA_CHECKED(D3D12_FEATURE_SHADER_CACHE, m_dShaderCache);
+
+        // Initialize per-node feature support data structures
+        const UINT uNodeCount = m_pDevice->GetNodeCount();
+        m_dProtectedResourceSessionSupport.resize(uNodeCount);
+        m_dArchitecture1.resize(uNodeCount);
+        for (UINT i = 0; i < uNodeCount; i++) {
+            INITIALIZE_MEMBER_DATA_CHECKED(D3D12_FEATURE_PROTECTED_RESOURCE_SESSION_SUPPORT, m_dProtectedResourceSessionSupport[i]);
+            INITIALIZE_MEMBER_DATA_CHECKED(D3D12_FEATURE_ARCHITECTURE1, m_dArchitecture1[i]);
+        }
+
+        return result;
+    }
+
+    HRESULT GetStatus() const
+    {
+        return m_hStatus;
+    }
+
+    static std::variant<CD3DX12FeatureSupport, HRESULT> Create(ID3D12Device* pDevice)
+    {
+        CD3DX12FeatureSupport features(pDevice);
+        HRESULT result = features.Init();
+        if (FAILED(result)) {
+            return result;
+        }
+
+        return features;
+    }
+
+    // 0: D3D12_OPTIONS
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, DoublePrecisionFloatShaderOps);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, OutputMergerLogicOp);
+    FEATURE_SUPPORT_GET(D3D12_SHADER_MIN_PRECISION_SUPPORT, m_dOptions, MinPrecisionSupport);
+    FEATURE_SUPPORT_GET(D3D12_TILED_RESOURCES_TIER, m_dOptions, TiledResourcesTier);
+    FEATURE_SUPPORT_GET(D3D12_RESOURCE_BINDING_TIER, m_dOptions, ResourceBindingTier);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, PSSpecifiedStencilRefSupported);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, TypedUAVLoadAdditionalFormats);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, ROVsSupported);
+    FEATURE_SUPPORT_GET(D3D12_CONSERVATIVE_RASTERIZATION_TIER, m_dOptions, ConservativeRasterizationTier);
+    FEATURE_SUPPORT_GET(UINT, m_dOptions, MaxGPUVirtualAddressBitsPerResource);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, StandardSwizzle64KBSupported);
+    FEATURE_SUPPORT_GET(D3D12_CROSS_NODE_SHARING_TIER, m_dOptions, CrossNodeSharingTier);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, CrossAdapterRowMajorTextureSupported);
+    FEATURE_SUPPORT_GET(BOOL, m_dOptions, VPAndRTArrayIndexFromAnyShaderFeedingRasterizerSupportedWithoutGSEmulation);
+    FEATURE_SUPPORT_GET(D3D12_RESOURCE_HEAP_TIER, m_dOptions, ResourceHeapTier);
+
+    // 1: Architecture (Deprecated)
+    // FEATURE_SUPPORT_GET_NODE_INDEXED(BOOL, m_dArchitecture, TileBasedRenderer);
+    // FEATURE_SUPPORT_GET_NODE_INDEXED(BOOL, m_dArchitecture, UMA);
+    // FEATURE_SUPPORT_GET_NODE_INDEXED(BOOL, m_dArchitecture, CacheCoherentUMA);
+
+    // 2: Feature Levels
+    // User inputs a list of feature levels. The function returns the highest level supported.
+    // Option 1: Straightforward implementation. Only perform forwarding
+    HRESULT HighestFeatureLevel(UINT NumFeatureLevels, const D3D_FEATURE_LEVEL* pFeatureLevelRequested, D3D_FEATURE_LEVEL& MaxSupportedFeatureLevel)

```

Comment:
Personally I think this should be cached during creation, and just find the highest feature level supported from the full list available in the header.